unit_test/main_tests/main_platform_test.py

```diff
@@ -191,3 +197,64 @@ def test_archs_platform_all(platform, intercepted_build_args, monkeypatch):
             Architecture.arm64,
             Architecture.universal2,
         }
+
+
+@pytest.mark.parametrize(
+    "only,plat",
+    (
+        ("cp311-manylinux_x86_64", "linux"),
+        ("cp310-win_amd64", "windows"),
+        ("cp311-macosx_x86_64", "macos"),
+    ),
+)
+def test_only_argument(monkeypatch, only, plat):
+    from cibuildwheel import linux, macos, windows
+
+    monkeypatch.setenv("CIBW_BUILD", "unused")
+    monkeypatch.setenv("CIBW_SKIP", "unused")
+    monkeypatch.setattr(sys, "argv", sys.argv + ["--only", only])
+
+    intercepted = ArgsInterceptor()
+
+    if plat == "linux":
+        monkeypatch.setattr(linux, "build", intercepted)
+    elif plat == "macos":
+        monkeypatch.setattr(macos, "build", intercepted)
+    elif plat == "windows":
+        monkeypatch.setattr(windows, "build", intercepted)
+
+    main()
+    options = intercepted.args[0]
+    assert options.globals.build_selector.build_config == only
+    assert options.globals.build_selector.skip_config == ""
+    assert options.platform == plat
+    assert options.globals.architectures == Architecture.all_archs(plat)
+
+
+@pytest.mark.parametrize("only", ("cp311-manylxinux_x86_64", "some_linux_thing"))
+def test_only_failed(intercepted_build_args, monkeypatch, only):
+    monkeypatch.setattr(sys, "argv", sys.argv + ["--only", only])
+    monkeypatch.delenv("CIBW_PLATFORM")
```

Comment:
I'm curious why these delenvs are needed? Also on line 244, and 254.

Reply:
Because `intercepted_build_args` uses `platform` which injects it. The point of this is to see if adding something like `--platform` or `--archs` fails or some broken `--only` is used, but instead it will fail because `CIBW_PLATFORM` is set & `--only` is present.

Reply:
I've removed the coupling of intercepted_build_args from platform - it now monkeypatches all three. I'll push that to this PR so you can have a look.